src/cameraGeom/TransformMap.cc

```diff
@@ -0,0 +1,171 @@
+/*
+ * LSST Data Management System
+ * Copyright 2014 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#include <exception>
+#include <memory>
+#include <sstream>
+#include <type_traits>
+
+#include "lsst/pex/exceptions.h"
+#include "lsst/afw/cameraGeom/TransformMap.h"
+
+namespace lsst {
+namespace afw {
+namespace cameraGeom {
+
+namespace {
+
+/**
+ * Represent a set of camera Transforms as a FrameSet.
+ *
+ * @tparam Map Any type satisfying the STL map API and mapping CameraSys to
+ *             shared_ptr<Transform>.
+ *
+ * @param root Common coordinate system serving as root of FrameSet's tree.
+ * @param transforms A map whose keys are coordinate systems, and whose
+ *                   values point to Transforms that convert from `root` to the
+ *                   corresponding key. All Transforms must be invertible.
+ * @return a newly allocated FrameSet corresponding to the method arguments
+ *
+ * @throws lsst::pex::exceptions::InvalidParameterError Thrown if
+ *         `transforms` contains `root`, or if any Transform is not
+ *         invertible.
+ */
```

Comment:
Make this a regular comment; it's private.  Same for other private functions in this file.